File: music-android-java/module_login/src/main/java/com/czb/module_login/model/LoginApi.java
package com.czb.module_login.model;

import com.czb.module_login.model.bean.LoginBean;
import com.czb.module_login.model.bean.User;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface LoginApi {

    @POST("/user/login/{captcha}/{captcha_key}")
    Call<LoginBean> login(@Path("captcha")String captcha ,
                          @Path("captcha_key")String captchaKey,
                          @Body User data);


}

File: music-android/module_login/src/main/java/com/chen/module_login/LoginApi.kt
package com.chen.module_login

import com.chen.module_base.net.BaseResponse
import com.chen.module_login.bean.LoginBean
import com.chen.module_login.bean.LoginInBean
import com.chen.module_login.bean.ResultData
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.Path

interface LoginApi {
    companion object{
        const val BASE_URL = "http://10.0.2.2:8090"
        // 登录
        const val LOGIN_URL = "${BASE_URL}/user/login"
    }

    @POST("/user/login/{captcha}/{captcha_key}")
    suspend fun login(
        @Path("captcha") captcha: String,
        @Path("captcha_key") captchaKey: String,
        @Body query: LoginInBean
    ): BaseResponse<String?>
}